api/src/services/googleapi/googleapi.js

```diff
@@ -0,0 +1,214 @@
+import { google } from 'googleapis'
+import fs from 'fs'
+
+const createGoogleDriveClient = async () => {
+  const auth = new google.auth.GoogleAuth({
+    credentials: {
+      client_email: process.env.CLIENT_EMAIL,
+      private_key: process.env.PRIVATE_KEY.replace(/\\n/gm, '\n'),
+    },
+    scopes: ['https://www.googleapis.com/auth/drive'],
+  })
+
+  const authClient = await auth.getClient()
+
+  const driveClient = google.drive({
+    version: 'v3',
+    auth: authClient,
+  })
+
+  return driveClient
+}
+
+const createGoogleSheetClient = async () => {
+  const auth = new google.auth.GoogleAuth({
+    credentials: {
+      client_email: process.env.CLIENT_EMAIL,
+      private_key: process.env.PRIVATE_KEY.replace(/\\n/gm, '\n'),
+    },
+    scopes: ['https://www.googleapis.com/auth/spreadsheets'],
+  })
+
+  const authClient = await auth.getClient()
+
+  const sheetClient = google.sheets({
+    version: 'v4',
+    auth: authClient,
+  })
+
+  return sheetClient
+}
+
+export const createSheetGoogleAPI = async () => {
+  const driveClient = await createGoogleDriveClient()
+
+  const sheet = await driveClient.files.create({
+    resource: {
+      name: 'Your SheetMenu',
+      mimeType: 'application/vnd.google-apps.spreadsheet',
+    },
+    media: {
+      mimeType:
+        'application/vnd.openxmlformats-officedocument.spreadsheetml.sheet',
+      body: fs.createReadStream('src/templates/sheetmenu-template.xlsx'),
+    },
+  })
+
+  if (!sheet?.data?.id) {
+    return
+  }
+
+  await driveClient.permissions.create({
+    resource: {
+      type: 'user',
+      role: 'writer',
+      emailAddress: context.currentUser.email,
+    },
+    fileId: sheet?.data?.id,
```

Comment:
We don't need `?` because we return on line 57 if it doesn't exist.

```suggestion
    fileId: sheet.data.id,
```